src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,21 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (options.ClientCertificates.Count > 0)
+            {
+                if (!MessageWebSocketClientCertificateSupported)
+                {
+                    throw new PlatformNotSupportedException(string.Format(
+                        CultureInfo.InvariantCulture,
+                        SR.net_WebSockets_UWPClientCertSupportRequiresWindows10InsiderPreviewBuild16215OrGreater));
+                }
+
+                RTCertificate winRtClientCert = ConvertDotNetClientCertToWinRtClientCert(options.ClientCertificates[0]);
```

Comment:
One thing to mention is that I see you are only taking the first certificate from the array of certs.  Did you ever wonder why there is an array of certs and not a parameter taking a single cert?

Historically, System.Net always took a collection of certificates because some of the client certs could have different issuers.  And the old networking stack would pass the trusted issuers list from the server during the network i/o.  So, System.Net would use that information and help filter down the collection to find client certs that matched the issuers.  And then it would take the first one from that.

By default, now, the Windows Server OS doesn't send a trusted issuer list for client cert negotiation. And since we're using WinRT APIs, we don't have visibility into that trusted issuers list anymore.

So, in general, we ended up always just using the first certificate.

But see this:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCertificateHelper.cs#L84

I do a little filtering on the list of certs to make sure they are client certs, etc. and then take the first one from that.